src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -220,12 +220,17 @@ private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage reques
 
                     if (streamingEnabled)
                     {
-                        Stream stream = await request.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(true);
-                        cancellationToken.ThrowIfCancellationRequested();
-
-                        ReadableStreamPullState pullState = new ReadableStreamPullState(stream, cancellationToken);
-
-                        promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, ReadableStreamPull, pullState);
+                        JSObject transformStream = BrowserHttpInterop.CreateTransformStream();
+                        try
+                        {
+                            promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, transformStream);
+                            promise = WasmHttpWriteStream.CopyToAsync(promise, transformStream, request.Content, cancellationToken);
```

Comment:
this second promise is not a proxy of the JS promise and so, it would not work with `BrowserHttpInterop.CancelationHelper`, it will assert/throw I think.